### DISCUSSION TITLE
Proposal to add Member role to Apollo Community

### DISCUSSION THREAD

**Discussion by nobodyiam** [apolloconfig/apollo#4239] [General]
According to [Apollo Governance Rule](https://github.com/apolloconfig/apollo/blob/master/GOVERNANCE.md), there are 4 roles in Apollo Community:
* Users
* Contributors
* Committers
* Project Management Committee (PMC)

This is a very clear community structure. However, I have a feeling that the gap between `Contributors` and `Committers` is a little big and we could add a `Member` role between them to make the transition more smooth.

The `Member` role is designed for those active contributors who have made multiple contributions to the project or community. By assigning them the `Member` roles, they will be invited as a [member](https://github.com/orgs/apolloconfig/people) of the apollo community and have a badge shown on their GitHub profile page. Hopefully, they would participate in community discussions more and remain active contributors to the community.

What do you think of this idea? Please feel free to share your ideas or suggestions.

> BTW, this idea is inspired by the [dapr community](https://github.com/dapr/community/blob/master/community-membership.md#member).

**Comment by zouyx** [comment]
I think that is great . But what's the responsibilities of `Member` role?

**Reply by nobodyiam** [reply]
The main role would be:
1. Active owner of code contributed by them, e.g. Addresses bugs or issues discovered after code is accepted
2. Responsive to issues and PRs assigned to them

**Reply by zouyx** [reply]
In my option,that's no problem. But do they have voting rights in review?

**Reply by nobodyiam** [reply]
According to [Requesting a pull request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review), I think the answer is yes.

**Reply by zouyx** [reply]
Thanks and no problem

**Comment by nisiyong** [comment]
If we introduce the Member role, I think we should make the role responsibilities more clearly, such as: 
* the difference between Member and Committer
* which role could merge the PR
* ...

**Reply by nobodyiam** [reply]
Good point. I think the member role is for those active contributors but not yet active enough to be a committer.
Here is a rough relationship in my mind:
> user -> contributor -> member -> committer

I believe it would be the committer to merge the PRs, but a member could help to review PRs and give their comments.
Specifically, the member would be assigned `Triage` permission while the committer would be assigned `Write` permission.
![image](https://user-images.githubusercontent.com/837658/156084311-a71dbab4-95ce-4d78-9a48-c3e80dbbd266.png)

**Comment by nobodyiam** [comment]
I have submitted #4265 to add the membership role, please help to review.
